lib/auth/apiserver.go

```diff
@@ -341,6 +349,208 @@ func (s *APIServer) deleteReverseTunnel(auth ClientI, w http.ResponseWriter, r *
 	return message(fmt.Sprintf("reverse tunnel %v deleted", domainName)), nil
 }
 
+type upsertTrustedClusterReq struct {
+	TrustedCluster json.RawMessage `json:"trusted_cluster"`
+}
+
+func (s *APIServer) enableTrustedCluster(auth ClientI, tc services.TrustedCluster, version string) error {
+	vtcr := ValidateTrustedClusterRequest{
+		Token: tc.GetToken(),
+		CAs:   [][]byte{},
+	}
+
+	for _, caType := range []services.CertAuthType{services.HostCA} {
+		ca, err := auth.GetCertAuthorities(caType, false)
+		if err != nil {
+			return trace.Wrap(err)
+		}
+
+		for _, sca := range ca {
+			data, err := services.GetCertAuthorityMarshaler().MarshalCertAuthority(sca, services.WithVersion(version))
```

Comment:
marshal/unmarshal should stay on apiserver level, while auth server should work with interfaces.